overlord/snapstate/snapstate.go

```diff
@@ -846,6 +846,14 @@ var kindRevOrder = map[snap.Type]int{
 }
 
 func (bk byKind) Less(i, j int) bool {
+	// snapd is special and sorts first
```

Comment:
Would be nice to point out why here in these two comments, so we can remember in the future (or right now even.. I can't remember why that's required :-).